lang/rust/Makefile

```diff
@@ -0,0 +1,132 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=rust
+
+PKG_VERSION:=1.62.0
+PKG_RELEASE:=1
+
+PKG_LICENSE:=Apache-2.0 MIT
+PKG_LICENSE_FILES:=LICENSE-APACHE LICENSE-MIT
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/rust-lang/rust.git
+PKG_SOURCE_DATE:=2022-06-29
+PKG_SOURCE_VERSION:=a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
+PKG_MIRROR_HASH:=09f933ee7a449751bc6d7ad6364dd44163439f029061daf158d7bfaf3a726184
+
+HOST_BUILD_PARALLEL:=1
+HOST_USE_NINJA:=1
+PKG_HOST_ONLY:=1
+HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/rust-$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/host-build.mk
+include ./rustc_environment.mk
+
+HOST_STAMP_PREPARED=$(HOST_BUILD_DIR)/.prepared_$(RUSTC_TARGET_ARCH)
+HOST_STAMP_BUILT=$(HOST_BUILD_DIR)/.built_$(RUSTC_TARGET_ARCH)
+HOST_STAMP_CONFIGURED=$(HOST_BUILD_DIR)/.configured_$(RUSTC_TARGET_ARCH)
+HOST_STAMP_INSTALLED=$(CARGO_HOME)/.rust_$(RUSTC_TARGET_ARCH)_installed
+
+# Requires Python3 to build
+HOST_BUILD_DEPENDS:=python3/host rust_host/host
+
+# Use External rust-lang specific LLVM
+RUST_UNINSTALL:=$(STAGING_DIR_HOST)/lib/rustlib/uninstall.sh
+RUST_TMP_DIR:=$(TMP_DIR)/rust-install
+
+# Set Configure Environmentals
+HOST_CONFIGURE_OPTS += CARGO_HOME="$(CARGO_HOME)"
+
+# The Flags are Common between HOST and TARGET
+RUST_COMMON_ARGS := \
+  --build=$(RUSTC_HOST_ARCH) \
+  --host=$(RUSTC_HOST_ARCH) \
+  --dist-compression-formats=xz \
+  --enable-llvm-link-shared \
+  --enable-llvm-plugins \
+  --enable-missing-tools \
+  --enable-ninja \
+  --enable-optimize \
+  --enable-optimize-llvm \
+  --enable-parallel-compiler \
+  --enable-sanitizers \
+  --release-channel=nightly \
+  --prefix=$(STAGING_DIR_HOST) \
+  --bindir=$(STAGING_DIR_HOST)/bin \
+  --libdir=$(STAGING_DIR_HOST)/lib \
+  --sysconfdir=$(STAGING_DIR_HOST)/etc \
+  --datadir=$(STAGING_DIR_HOST)/share \
+  --mandir=$(STAGING_DIR_HOST)/man \
+  --set=build.extended=true \
+  --llvm-root=$(LLVM_DIR) \
+  --set=target.$(RUSTC_TARGET_ARCH).llvm-config=$(LLVM_DIR)/bin/llvm-config
+
+# MUSL Targets need additional flags
+MUSL_ARGS = \
+  --set=target.$(RUSTC_TARGET_ARCH).ar=$(TARGET_AR) \
+  --set=target.$(RUSTC_TARGET_ARCH).cc=$(TARGET_CC_NOCACHE) \
+  --set=target.$(RUSTC_TARGET_ARCH).cxx=$(TARGET_CXX_NOCACHE) \
+  --set=target.$(RUSTC_TARGET_ARCH).linker=$(TARGET_CC_NOCACHE) \
+  --set=target.$(RUSTC_TARGET_ARCH).musl-root=$(TOOLCHAIN_DIR) \
+  --set=target.$(RUSTC_TARGET_ARCH).ranlib=$(TARGET_RANLIB)
+
+##
+# Because OpenWrt uses default configure flags that are not recognized
+# we override the default and set our own.
+#
+# Rust HOST Configuration Arguments
+HOST_CONFIGURE_ARGS = \
+  --target=$(RUSTC_TARGET_ARCH) \
+  ${RUST_COMMON_ARGS} \
+  $(MUSL_ARGS) \
+  --set=build.cargo=$(STAGING_DIR_HOST)/bin/cargo \
+  --set=build.rustc=$(STAGING_DIR_HOST)/bin/rustc
```

Comment:
this throws an error while building rust
```--set=build.rustc=$(STAGING_DIR_HOST)/bin/rustc```

``````error: attributes starting with `rustc` are reserved for use by the `rustc` compiler     --> library/core/src/lib.rs:87:4```